### PR TITLE
Add Range

### PR DESCRIPTION
**Pull request by julienrf**
I also added NumericRange but we should explore (later) a way to unify `Range` and `NumericRange`.

### CODE REVIEW

File: src/main/scala/strawman/collection/immutable/Range.scala [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+  *  @param step        the step for the range.
+  *  @param isInclusive whether the end of the range is included or not
+  */
+@SerialVersionUID(7618862778670199309L)
+final class Range(
+  val start: Int,
+  val end: Int,
+  val step: Int,
+  val isInclusive: Boolean
+)
```
Comment (julienrf) [inline src/main/scala/strawman/collection/immutable/Range.scala]: The implementation is essentially copy-pasted from the current collections, with a few differences. For instance, there is no more `Inclusive` subtype of `Range`.

**Comment by pathikrit** [comment]
Is there a way to make Range generic e.g. `Range[A](implicit succ: Succesor[A])`? This way we can define `Range[sql.Date]` etc.

**Comment by Ichoran** [comment]
@pathikrit - No, that's definitely `NumericRange` territory except without even a numeric representation.  `Range` is for integers (since you need something efficient to cover integers, and the generic implementations have not been efficient enough).

**Comment by Ichoran** [comment]
Belated, LGTM

**Pull request merged**